libs/entitlement/src/main/java/org/elasticsearch/entitlement/runtime/policy/entitlements/FilesEntitlement.java

```diff
@@ -120,12 +120,10 @@ public int hashCode() {
         }
 
         static FileData ofPath(Path path, Mode mode) {
-            assert path.isAbsolute();
```

Comment:
Why do we need to remove these asserts? We can specify absolute and relative paths in Windows. I don't think we should add leniency to make windows testing easier, let's adjust the tests so they aren't assuming `/` is the root path.

Reply:
It's in the description; the tests are fixed, but the asserts will fail anyway in production code because we have unix paths in policies with a cluster running on windows and vice-versa.

Reply:
I think we can solve the eg `/proc/meminfo` in a different way: we don't need to add those to the policy on windows. Relaxing the assertions make it unclear why `path` and `relative_path` exist when there is no enforcement on what exists there.

Reply:
There will always be the matter of policy files; unless we add platform-specific sections to them, we may end up with a mix of absolute paths for different platforms.
I don't think we want that complication (platform-specific sections); if we do, we can address this separately.